src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -125,3 +127,227 @@ def compute_reference(frames: np.ndarray,
         )
 
     return ref_image
+
+
+def optimize_motion_parameters(initial_frames: np.ndarray,
+                               suite2p_args: dict,
+                               logger: callable = None) -> dict:
+    """Loop over a range of parameters and select the best set from the
+    max acutance of the final, average image.
+
+    Parameters
+    ----------
+    initial_frames : numpy.ndarray, (N, M, K)
+        Smaller subset of frames to create a reference image from.
+    suite2p_args : dict
+        A dictionary of suite2p configs containing at minimum:
+
+        ``"h5py"``
+            HDF5 file containing to the movie to motion correct.
+        ``"h5py_key"``
+            Name of the dataset where the movie to be motion corrected is
+            stored.
+        ``"maxregshift"``
+            Maximum shift allowed as a fraction of the image dimensions.
+        ``"smooth_sigma_min"``
+            Minimum spatial Gaussian smoothing parameter used by suite2p to
+            smooth frames before correlation. Used as minimum start value in
+            parameter search (float).
+        ``"smooth_sigma_max"``
+            Maximum value for allowed search in smooth_sigma (float).
+        ``"smooth_sigma_steps"``
+            Number of steps to take between min and max in parameter search
+            (int).
+        ``"smooth_sigma_time_min"``
+            Minimum time Gaussian smoothing of frames to apply before
+            correlation. Used as minimum start value in parameter search
+            (float).
+        ``"smooth_sigma_time_max"``
+            Maximum value for allowed search in smooth_sigma (float).
+        ``"smooth_sigma_time_steps"``
+            Number of steps to take between min and max in parameter search
+            (int).
+    logger : callable, optional
+        Function to print to stdout or a log.
+
+    Returns
+    -------
+    best_result : dict
+        A dict containing the final results of the search:
+
+        ``ave_image``
+            Image created with the settings yielding the highest image acutance
+            (numpy.ndarray, (N, M))
+        ``ref_image``
+            Reference Image created with the settings yielding the highest
+            image acutance (numpy.ndarray, (N, M))
+        ``acutance``
+            Acutance of ``best_image``. (float)
+        ``smooth_sigma``
+            Value of ``smooth_sigma`` found to yield the best acutance (float).
+        ``smooth_sigma_time``
+            Value of ``smooth_sigma_time`` found to yield the best acutance
+            (float).
+    """
```

Comment:
I'm not terribly happy that we are adopting Suite2Ps "all the args in a dict" API, but I acknowledge that that is the simplest thing to do given that this code depends on `register_frames`. Can you please add a check at the top of this method to make sure all of the required keys are present in `suite2p_args` so that we don't get an unexpected `KeyError` during a compute step.

Reply:
Are the doc string descriptions not enough here or in the next function? Where we are at in the pipeline, all of these settings should exist, many of which are required by the pipeline schemas. I could modify `add_required_parameters` to do this job if we still want it.

Reply:
I'm being a little absurd and imagining a case where, someday, we are doing some debugging work and calling this method from outside of `suite2p_registration/__main__.py`.

I would like to have that option just in case.

Reply:
I guess I'm just railing against Suite2P's API, which isn't our fault.

No need to make this change.